live-ingester/src/integration-test/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/response/ATSTransformResponseHandlerTest.java

```diff
@@ -0,0 +1,244 @@
+    {
+        // given
+        TransformResponse transformResponse = new TransformResponse(
+                "e5bcc533-853c-44f1-a02d-3ab1e36a03e6",
+                new ClientData(
+                        "dcc9e89b-c1c0-48a4-814f-98eafab72fdb", "application/pdf", 0),
+                400,
+                "Something went wrong!");
+
+        // when
+        simulateResponse(transformResponse);
+
+        // then
+        then(ingestContentCommandHandler).shouldHaveNoInteractions();
+    }
+
+    @Test
+    void shouldRetryRouteGivenNumberOfTimeOnAnyNonTransformRelatedError()
+    {
+        // given
+        String nodeId = "dcc9e89b-c1c0-48a4-814f-98eafab72fdb";
+        String transformedFileId = "e5bcc533-853c-44f1-a02d-3ab1e36a03e6";
+
+        TransformResponse transformResponse = new TransformResponse(
+                transformedFileId,
+                new ClientData(
+                        nodeId, "application/pdf", 0),
+                202,
+                "");
+
+        IngestContentCommand expectedCommand = new IngestContentCommand(
+                transformedFileId,
+                nodeId);
+
+        doThrow(new LiveIngesterRuntimeException("Some exception")).when(ingestContentCommandHandler).handle(expectedCommand);
+
+        // when
+        try
+        {
+            simulateResponse(transformResponse);
+        }
+        catch (CamelExecutionException ignored)
+        {}
+
+        // then
+        then(ingestContentCommandHandler).should(times(retryIngestion.attempts() + 1)).handle(expectedCommand);
+    }
+
+    @ParameterizedTest
+    @ValueSource(classes = {EmptyRenditionException.class, ResourceNotFoundException.class})
+    void shouldRetryTransformationOnSpecificExceptions(Class<? extends Throwable> exception)
+    {
+        // given
+        String nodeId = "dcc9e89b-c1c0-48a4-814f-98eafab72fdb";
+        String transformedFileId = "e5bcc533-853c-44f1-a02d-3ab1e36a03e6";
+        String targetMimeType = "application/pdf";
+        int retryAttempt = 1;
+
+        TransformResponse transformResponse = new TransformResponse(
+                transformedFileId,
+                new ClientData(
+                        nodeId, targetMimeType, 0),
+                202,
+                "");
+
+        IngestContentCommand expectedCommand = new IngestContentCommand(
+                transformedFileId,
+                nodeId);
+
+        doThrow(exception).when(ingestContentCommandHandler).handle(expectedCommand);
+
+        TransformRequest expectedTransformRequest = new TransformRequest(nodeId, targetMimeType);
+
+        // when
+        try
+        {
+            simulateResponse(transformResponse);
+        }
+        catch (CamelExecutionException ignored)
+        {}
+
+        // then
+        then(ingestContentCommandHandler).should(times(1)).handle(expectedCommand);
+        then(atsTransformRequester).should().requestTransformRetry(expectedTransformRequest, retryAttempt);
```

Comment:
You could put this in the catch block and have a `fail` statement if no exception was received.

Reply:
in real scenario this error is not thrown as I believe onException() is catching it and I'm not sure why it is thrown here so I would leave it as it is